### PR TITLE
Refactoring part 4

### PR DESCRIPTION
**Pull request by IvanGoncharov**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -439,6 +447,12 @@ Converter.prototype.buildParameter = function(oldParameter) {
       'Complex type is used inside non-body argument.');
   }
 
+  //According to Swagger 2.0 spec: If the parameter is in "path",
+  //this property is required and its value MUST be true.
+  if (parameter.in === 'path') {
```
Comment (mohsen1) [inline index.js]: @webron do you have to specify `required` for path parameters? I see schema validates a spec that has a path parameter without `required` property. 
@IvanGoncharov if it's optional, it's probably a good idea to explicitly specify it here.

File: index.js [hunk 2]
```diff
@@ -259,12 +260,32 @@ Converter.prototype.buildTypeProperties = function(oldType) {
...
-  var result = this.buildTypeProperties(
-    oldDataType.type || oldDataType.dataType || oldDataType.responseClass
-  );
+  var oldType =
+    oldDataType.type || oldDataType.dataType || oldDataType.responseClass;
+  var oldItems = oldDataType.items;
+
+  if (isValue(oldType)) {
+    //handle List[<TYPE>] types from 1.1 spec
+    var match = oldType.match(/^(.*)\[(.*)\]$/);
```
Comment (mohsen1) [inline index.js]: Can you please describe this regular expression?

Comment (IvanGoncharov) [inline index.js]: I update comment message.
It pretty simple RegEx, see:
http://regexper.com/#%5E(.*)%5C%5B(.*)%5C%5D%24

File: index.js (hunk 1 above)
Comment (webron) [inline index.js]: @mohsen1 - yes, you do. The schema also requires it - https://github.com/swagger-api/swagger-spec/blob/master/schemas/v2.0/schema.json#L774-L778.

Comment (mohsen1) [inline index.js]: @webron the `required` property is not required. This is valid in editor at least:


```yaml
---
swagger: '2.0'
info:
  version: 0.0.0
  title: Simple API
paths:
  /{foo}:
    parameters:
      - name: foo
        in: path
        type: string
      
    get:
      responses:
        200:
          description: OK
```

**Comment by mohsen1** [comment]
Thank you!

File: index.js (hunk 1 above)
Comment (webron) [inline index.js]: See, that what happens when I don't spend enough time on things :) Anyways, right you are, I'll open a ticket on swagger-spec to fix it in the schema. It *is* required though, just a bug in the schema.

**Pull request merged**